### PR TITLE
[Feature] RRandomCrop

### LINKED ISSUE #122: Support Rotated RandomCrop
## Description

This issue describes the development requirement of a feature we wish to collaborate with the community.
In this issue, we welcome the community developers to participate in the development of `RandomCrop`.

To inherit a new `RandomCrop` in MMRotate to support OBBs.

## Steps

- [ ] Implement the `RotatedRandomCrop`
- [ ] Add unit tests to test your code

## Resources
  - https://github.com/open-mmlab/mmdetection/blob/e045607843f439f064791506257e9b4fdbf268c0/mmdet/datasets/pipelines/transforms.py#L712

### PR DESCRIPTION
**Pull request by nijkah**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Please describe the motivation of this PR and the goal you want to achieve through this PR.
closes #122 

## Modification
Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)
Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)
If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmrotate/datasets/pipelines/transforms.py [hunk 1]
```diff
@@ -262,3 +262,99 @@ def __repr__(self):
...
         return repr_str
+
+
+@ROTATED_PIPELINES.register_module()
+class RRandomCrop(RandomCrop):
+    """Random crop the image & bboxes.
+
+    The absolute `crop_size` is sampled based on `crop_type` and `image_size`,
+    then the cropped results are generated.
+    Args:
```
Comment (zytx121) [inline mmrotate/datasets/pipelines/transforms.py:273]: A blank line is required in front of this line.

File: mmrotate/datasets/pipelines/transforms.py [hunk 2]
```diff
@@ -262,3 +262,99 @@ def __repr__(self):
...
+            (h * crop_size[0], w * crop_size[1]) part from an input of size
+            (h, w). "relative_range" uniformly samples relative crop size from
+            range [crop_size[0], 1] and [crop_size[1], 1] for height and width
+            respectively. "absolute" crops from an input with absolute size
+            (crop_size[0], crop_size[1]). "absolute_range" uniformly samples
+            crop_h in range [crop_size[0], min(h, crop_size[1])] and crop_w
+            in range [crop_size[0], min(w, crop_size[1])]. Default "absolute".
+        allow_negative_crop (bool, optional): Whether to allow a crop that does
+            not contain any bbox area. Default False.
+    Note:
```
Comment (zytx121) [inline mmrotate/datasets/pipelines/transforms.py:287]: A blank line is required in front of this line.

File: mmrotate/datasets/pipelines/transforms.py [hunk 3]
```diff
@@ -262,3 +262,99 @@ def __repr__(self):
...
+
+    def _crop_data(self, results, crop_size, allow_negative_crop):
+        """Function to randomly crop images, bounding boxes.
+
+        Args:
+            results (dict): Result dict from loading pipeline.
+            crop_size (tuple): Expected absolute size after cropping, (h, w).
+            allow_negative_crop (bool): Whether to allow a crop that does not
+                contain any bbox area. Default to False.
+        Returns:
```
Comment (zytx121) [inline mmrotate/datasets/pipelines/transforms.py:314]: A blank line is required in front of this line.

**Review by liuyanyi — COMMENTED** [review]
Great job.
I have a little question, if an object was cropped and the center is within the boundary, should we consider the outer area? 
image

**Comment by nijkah** [comment]
@liuyanyi That's a quite difficult question. Yes, I also expect that whether or not to clip the outer area of the object affects training a model in a certain way. Honestly, I have no idea unless experiment with it.

I thought it is not necessary because `bbox_clip_border` is optional parameter in `RandomCrop` and some other implementations.

**Review by zytx121 — APPROVED** [review]
LGTM

**Review by liuyanyi — APPROVED** [review]

**Pull request merged**